Improve command descriptions

hearvox opened the pull request:
Also: 
* Match WP Code Ref grammar for headings: Cap-word (instead of CAP).
* Add Codex or Handbook links to some commands.
Done for WordCamp for Publishers: Contributor Day 2017.

Note: Something in the CLI Handbook's existing Examples syntax is causing page displays to break -- sometimes. Will do some testing.

danielbachhuber commented:
Hi @hearvox,

Thanks for the pull request.

This documentation is generated dynamically from the WP-CLI codebase. In order to change it, you'll need to submit pull requests to each corresponding part of the codebase.

From there, we'll be able to run the script that then generates the documentation.

gitlost commented:
Great stuff, but I don't know that Cap-word instead of CAP applies, as these are special purpose markers (also there's BC considerations).

hearvox commented:
@gitlost, Thanks for the info. Be glad to change headings back to CAPs. 

My reasoning for Cap-word headings: The CLI Handbook's headings for top-level commands are "Options", "Examples", and "Global Parameters".

These become HTML h3 headings in the Handbook:
https://developer.wordpress.org/cli/commands/server/

This is kinda like "Parameters" h3 headings in the Code Ref handbook:
https://developer.wordpress.org/reference/functions/add_meta_box/

Usually best to have the same typography for the same tags. Different typographies imply a difference to the user -- not sure that's your intent for CLI Handbook headings.

Caveat: I don't know what "BC" stands for, so maybe that trumps consistent typography. If so, I'll change them back and re-submit (per @danielbachhuber 's instruction). thanks.

danielbachhuber commented:
I'd prefer to keep the headings as CAP in the source code. We can transform them to proper case in the documentation generation process.

gitlost commented:
@hearvox sorry BC is for "backward compatible".

hearvox commented:
@danielbachhuber, I could use a little more info on how to proceed. A couple of us Docs folk were asked (at WordCamp for Publishers) to improve the top-level command descriptions in the CLI Handbook:
https://developer.wordpress.org/cli/commands/

You sent me (on Slack) to this 'handbook' repo link for instructions on updating CLI docs:
https://github.com/wp-cli/handbook/#wp-clihandbook

Those instructions say the cron for developer.wordpress.org/cli uses this DevHub_CLI class:
https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/themes/pub/wporg-developer/inc/cli.php

It pulls .md files listed in this JSON file:
https://raw.githubusercontent.com/wp-cli/handbook/master/bin/commands-manifest.json

Those .md files are the ones in this 'handbook' repo (to which I submitted a pull request). E.g., the current description for `wp sidebar` is "Manage sidebars", located in:
https://github.com/wp-cli/handbook/blob/master/commands/sidebar.md

The DevHub_CLI class pulls the first line of the that description into the excerpt of the docs post, which displays here:
https://developer.wordpress.org/cli/commands/

And it pulls the entire .md text into the post content:
https://developer.wordpress.org/cli/commands/sidebar/

But because `wp sidebar` command doesn't really manage sidebars, in my fork, I changed the first line to "Lists registered sidebars" (and added another line with a link to Sidebars in the Theme Handbook.)
https://github.com/hearvox/handbook/blob/master/commands/sidebar.md

Then submitted this pull request. But if the source of truth is not /wp-cli/handbook/blob/master/commands/sidebar.md, could you please point me to the repo/file to which is.

Sorry to be so dense but I'm not that github savvy. We have all the improved descriptions written, but still confused how to get them to developer.wordpress.org/cli/.

danielbachhuber commented:
> But if the source of truth is not /wp-cli/handbook/blob/master/commands/sidebar.md, could you please point me to the repo/file to which is.

For `wp sidebar`, this is: https://github.com/wp-cli/widget-command/blob/master/src/Sidebar_Command.php#L4

In each [command repo](https://github.com/wp-cli?utf8=%E2%9C%93&q=-command&type=&language=) you'll find a `src/` directory with one or more classes for the commands.

The GitHub icon for each top-level command doc will take you to the correct repo:

![image](https://user-images.githubusercontent.com/36432/29722885-59b1d8c8-8977-11e7-8802-636ab1891b1f.png)

By the way, you _almost_ connected all of the dots. The `.md` files are generated from the source code using a `wp handbook gen-commands` command.

hearvox commented:
@danielbachhuber, thanks so much. I feel like I reached the pot of gold at the end of a treasure hunt.

Turns out those are the same files I was working on with @ako80218, not realizing changes there got reflected at CLI Docs. Before the fork/edit/pull he was asked to submit each proposed change as an issue, so that's next: https://github.com/wp-cli/wp-cli/issues/4302

Long, hard, climb, but nearly there. Thanks for the clarity.

danielbachhuber commented:
Closing in favor of https://github.com/wp-cli/wp-cli/issues/4302
